backend/src/main/java/com/jobwait/security/LinkedInOAuthValidator.java

```diff
@@ -1,11 +1,22 @@
 package com.jobwait.security;
 
+import org.springframework.http.HttpStatus;
+
+import com.jobwait.fault.FaultException;
+
 public class LinkedInOAuthValidator extends OAuthValidator {
     @Override
-    public void validateToken(AuthToken token) {
+    public void validateToken(AuthToken token) throws FaultException {
         if (!token.clientId().startsWith("valid")) {
-            throw new RuntimeException("authtoken doesn't start with the word 'valid', therefore invalid");
+            throw new InvalidTokenFault();
         }
         // todo - query linkedin oauth endpoint and check the token
     }
+
+    private class InvalidTokenFault extends FaultException {
+        public InvalidTokenFault() {
+            super(HttpStatus.UNAUTHORIZED, "INVALID_AUTH_TOKEN",
+                    "authtoken doesn't start with the word 'valid', therefore invalid");
```

Comment:
i think you can remove the 'therefore invalid' part its in the error code/name as well, it will also be implicit when you send the error message without it.

Reply:
You're not wrong, but this message is going to change once we implement oauth2 stuff, so I'll leave it for now